CONTRIBUTING.md

```diff
@@ -0,0 +1,37 @@
+# Contribution guidelines
+
+First of all, thanks for thinking of contributing to this project. :smile:
+
+Before sending a Pull Request, please make sure that you're assigned the task on a GitHub issue.
+
+- If a relevant issue already exists, discuss on the issue and get it assigned to yourself on GitHub.
+- If no relevant issue exists, open a new issue and get it assigned to yourself on GitHub.
+
+Please proceed with a Pull Request only after you're assigned. It'd be sad if your Pull Request (and your hardwork) isn't accepted just because it isn't idealogically compatible.
+
+# Developing the gem
+
+1. Install the extensions dependencies : Mongo, R and Redis. For any issue related to these, kindly refer to the steps mentioned in the `.travis.yml` file.
```

Comment:
As those dependencies are not always easy to install, maybe we can soften this requirement? Like, "if you want to run entire specs suite... But to work on some particular feature..." or something like that.